pkg/accounts/kubernetes/validator.go

```diff
@@ -119,26 +155,54 @@ func makeClientFromSettings(ctx context.Context, settings map[string]interface{}
 		if err != nil {
 			return nil, err
 		}
-		return clientcmd.NewDefaultClientConfig(*cfg, makeOverrideFromAuthSettings(cfg, aSettings)), nil
+		return clientcmd.NewDefaultClientConfig(*cfg, makeOverrideFromAuthSettings(cfg, aSettings)).ClientConfig()
 	}
 	return nil, noValidKubeconfigError
 }
 
-func makeClientFromServiceAccount(ctx context.Context, spinSvc v1alpha2.SpinnakerServiceInterface) (clientcmd.ClientConfig, error) {
-	_, err := spinSvc.GetSpinnakerConfig().GetServiceSettingsPropString(ctx, util.ClouddriverName, "kubernetes.serviceAccountName")
+func makeClientFromServiceAccount(ctx context.Context, spinSvc v1alpha2.SpinnakerServiceInterface, c client.Client) (*rest.Config, error) {
```

Comment:
That'd make more sense to look for a SpinnakerService here instead of at the beginning.